tag/map_codec.go

```diff
@@ -127,8 +127,8 @@ func (eg *encoderGRPC) readBytesWithVarintLen() ([]byte, error) {
 
 	valueStart += eg.readIdx
 	valueEnd := valueStart + int(length)
-	if valueEnd > len(eg.buf) || length < 0 {
```

Comment:
True, length is a uint64 and the signature for binary.Uvarint says it'll be zero on any error so that's a redundant check.